### PR TITLE
Adopt TRT 10.0 on Paddle-TRT [1/3]

### PR DESCRIPTION
**Pull request by leo0519**
### PR Category

Others

### PR Types

Bug fixes

### Description

TRT 10.0 removed several deprecated APIs.
This MR supports Paddle-TRT with the compatibility between TRT 8.0 ~ 10.0.
- Replace deprecated APIs
- Add macro for previous version of TRT

### CODE REVIEW

**Comment by leo0519** [comment]
- PR-CI-APPROVAL: 
```
2024-04-23 08:40:15 0. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, HYBRID, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting parameter SERIAL, or setting TIMEOUT properties.
2024-04-23 08:40:15 The corresponding lines are as follows:
2024-04-23 08:40:15 test/ir/inference/CMakeLists.txt
2024-04-23 08:40:15 +    set_tests_properties(test_seq_concat_fc_fuse_pass PROPERTIES TIMEOUT 200)
2024-04-23 08:40:15 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
2024-04-23 08:40:15 There are 1 approved errors.
```
- PR-CI-Coverage: should be exempted (TRT version in CI < 8.6)

**Review by XieYunshen — DISMISSED** [review]
LGTM for `set_tests_properties(test_seq_concat_fc_fuse_pass PROPERTIES TIMEOUT 200)`

File: test/ir/inference/CMakeLists.txt [hunk 1]
```diff
@@ -158,7 +158,6 @@ if(WITH_GPU AND TENSORRT_FOUND)
   set_tests_properties(test_trt_ops_fp32_mix_precision PROPERTIES TIMEOUT 300)
   set_tests_properties(test_trt_inference_fp16_io PROPERTIES TIMEOUT 500)
   set_tests_properties(test_trt_convert_unary PROPERTIES TIMEOUT 600)
-  set_tests_properties(test_seq_concat_fc_fuse_pass PROPERTIES TIMEOUT 200)
```
Comment (Wangzheee) [inline test/ir/inference/CMakeLists.txt:161]: 请问这个单测在 trt后端的耗时会减小吗？onednn后端耗时会增加吗？

**Review by vivienfanghuagood — APPROVED** [review]

File: test/ir/inference/CMakeLists.txt (hunk 1 above)
Comment (leo0519) [reply test/ir/inference/CMakeLists.txt:161]: 這個單測在 `WITH_ONEDNN` 時新增，但 set_tests_properties 時並未確認 `WITH_ONEDNN` 是否為 True，因此將這行移至 `if(WITH_ONEDNN)` 底下

**Comment by leo0519** [comment]
Add new commits.

**Review by Wangzheee — DISMISSED** [review]
LGTM

File: paddle/fluid/inference/tensorrt/convert/nearest_interp_v2_op.cc [hunk 2]
```diff
@@ -44,7 +44,14 @@ class NearestInterpolateV2OpConverter : public OpConverter {
     auto out_w = PADDLE_GET_CONST(int, op_desc.GetAttr("out_w"));
 
     auto layer = TRT_ENGINE_ADD_LAYER(engine_, Resize, *input);
+#if IS_TRT_VERSION_GE(8600)
+    if (align_corners) {
```
Comment (Wangzheee) [inline paddle/fluid/inference/tensorrt/convert/nearest_interp_v2_op.cc:48]: 这个属性的使用改动，似乎没有单测验证

**Comment by leo0519** [comment]
PR-CI-APPROVAL:
```sh
2024-05-09 08:50:38 ****************
2024-05-09 08:50:38 0. You must have raindrops2sea or XiaoguangHu01 approval for change 20+ files or add than 1000+ lines of content.
2024-05-09 08:50:38 1. The error message you wrote in PADDLE_ENFORCE{_**} or PADDLE_THROW does not meet our error message writing specification. Possible errors include 1. the error message is empty / 2. the error message is too short / 3. the error type is not specified. Please read the specification [ https://github.com/PaddlePaddle/Paddle/wiki/Paddle-Error-Message-Writing-Specification ], then refine the error message. If it is a mismatch, please request chenwhql (Recommend), luotao1 or lanxianghit or Aurelius84 review and approve.
2024-05-09 08:50:38 The PADDLE_ENFORCE{_**} or PADDLE_THROW entries that do not meet the specification are as follows:
2024-05-09 08:50:38 PADDLE_ENFORCE_NOT_NULL(matmul_layer, + "TRT matrixmultiply layer building failed.")); 
2024-05-09 08:50:38 PADDLE_ENFORCE_NOT_NULL(add_layer, + "TRT elementwise add layer building failed.")); 
2024-05-09 08:50:38 2. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, HYBRID, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting parameter SERIAL, or setting TIMEOUT properties.
2024-05-09 08:50:38 The corresponding lines are as follows:
2024-05-09 08:50:38 test/ir/inference/CMakeLists.txt
2024-05-09 08:50:38 +    set_tests_properties(test_seq_concat_fc_fuse_pass PROPERTIES TIMEOUT 200)
2024-05-09 08:50:38 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
2024-05-09 08:50:38 There are 3 approved errors.
2024-05-09 08:50:38 ****************
```
PR-CI-Coverage: should be exempted (TRT version in CI < 8.6)

**Review by Wangzheee — COMMENTED** [review]
LGTM

File: paddle/fluid/inference/tensorrt/convert/nearest_interp_v2_op.cc (hunk 2 above)
Comment (leo0519) [reply paddle/fluid/inference/tensorrt/convert/nearest_interp_v2_op.cc:48]: 原先的 UT 便沒有覆蓋到，之後會新增相關 UT 到本 PR 做測試

**Review by Wangzheee — APPROVED** [review]

**Comment by leo0519** [comment]
- PR-CI-Coverage: This should be exempted since the original UTs do not cover all of op converters in Paddle-TRT.
- PR-CI-APPROVAL:

```sh
2024-05-24 08:49:45 0. You must have raindrops2sea or XiaoguangHu01 approval for change 20+ files or add than 1000+ lines of content.
2024-05-24 08:49:45 1. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, HYBRID, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting parameter SERIAL, or setting TIMEOUT properties.
2024-05-24 08:49:45 The corresponding lines are as follows:
2024-05-24 08:49:45 test/ir/inference/CMakeLists.txt
2024-05-24 08:49:45 +    set_tests_properties(test_seq_concat_fc_fuse_pass PROPERTIES TIMEOUT 200)
2024-05-24 08:49:45 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
2024-05-24 08:49:45 There are 2 approved errors.
```

**Review by qingqing01 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Pull request merged**